File: WEB/PHP/week_3/models/phones/Phone.php
<?php

namespace app\models\phones;

class Phone{
	private $id;
	private $brand;
	private $model;
	private $price;
	private $imageFileName;
	public function __construct($id,
								$brand,
								$model,
								$price,
								$imageFileName){
		$this->id = $id;
		$this->brand = $brand;
		$this->model = $model;
		$this->price = $price;
		$this->imageFileName = $imageFileName;
	}
	public function getId(){ return $this->id; }

	public function getBrand(){ return $this->brand; }

	public function getModel(){ return $this->model; }

	public function getPrice(){ return $this->price; }

	public function getImageFileName(){ return $this->imageFileName; }

	public function getPathToImage(){ return '../views/phones/images/' . $this->imageFileName; }

	public function setId($id){ $this->id = $id; }

	public function setBrand($brand){ $this->brand = $brand; }

	public function setModel($model){ $this->model = $model; }

	public function setPrice($price){ $this->price = $price; }

	public function setImageFile($imageFileName){ $this->imageFileName = $imageFileName; }
}

File: Java/Android/PhonesList/app/src/main/java/localhost/myapplication/models/Phone.java
package localhost.myapplication.models;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class Phone {
    private String brand;
    private String model;
    private Double price;

    public Phone(){}
    public Phone(String brand, String model, Double price){
        this.brand = brand;
        this.model = model;
        this.price = price;
    }

    public String getBrand() { return brand; }
    public void setBrand(String brand) { this.brand = brand; }
    public String getModel() { return model; }
    public void setModel(String model) { this.model = model; }
    public Double getPrice() { return price; }
    public void setPrice(Double price) { this.price = price; }

    @Override
    public boolean equals(@Nullable Object obj) {
        Phone p = (Phone)obj;
        return p.getBrand() == this.getBrand() &&
                p.getModel() == this.getModel() &&
                p.getPrice() == this.getPrice();
    }

    @NonNull
    @Override
    protected Object clone() throws CloneNotSupportedException {
        return new Phone(this.brand, this.model, this.price);
    }
}
